lib/matplotlib/axes/_axes.py

```diff
@@ -868,9 +868,9 @@ def _to_points(xy1, xy2, slope):
         if line.get_clip_path() is None:
             line.set_clip_path(self.patch)
         if not line.get_label():
-            line.set_label(f"_line{len(self.lines)}")
-        self.lines.append(line)
-        line._remove_method = self.lines.remove
+            line.set_label(f"_line{len(self._children)}")
```

Comment:
This changes the auto-generated label number. Do we care?

If so, we need to count the lines in children. If not we should do

```suggestion
            line.set_label(f"_child{len(self._children)}")
```

Reply:
I don't mind changing the name, but I don't want to iterate the list again just to count how many lines there.

Reply:
Then, let's change the name.

Reply:
If we are going to change this one to `_child` then we should change all of them (there is at least a `_image{N}`).